test/extended/util/disruption/frontends/known_backends.go

```diff
@@ -4,55 +4,109 @@ import (
+	if available {
+		if err := createOAuthRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+			return err
+		}
+		if err := createOAuthRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
+			return err
+		}
 	}
 
-	// Some jobs explicitly disable the console and other features. Check if it's disabled and if so,
-	// do not run a disruption monitoring backend for it.
-	configClient, err := configclient.NewForConfig(clusterConfig)
+	configAvailable, err := isConfigAPIGroupAvailable(clusterConfig)
 	if err != nil {
 		return err
 	}
-	clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
-	if err != nil {
-		e2e.Failf("Failed to get cluster version: %v", err)
-	}
-	// If the cluster does not know about the Console capability, it likely predates 4.12 and we can assume
-	// it has it by default. This is to catch possible future scenarios where we upgrade 4.11 no cap to 4.12 no cap.
-	if !cluster.KnowsCapability(clusterVersion, "Console") ||
-		cluster.HasCapability(clusterVersion, "Console") {
-		if err := createConsoleRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+	if configAvailable {
+		// Some jobs explicitly disable the console and other features. Check if it's disabled and if so,
+		// do not run a disruption monitoring backend for it.
+		configClient, err := configclient.NewForConfig(clusterConfig)
+		if err != nil {
 			return err
 		}
-		if err := createConsoleRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
-			return err
+		clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+		if err != nil {
+			e2e.Failf("Failed to get cluster version: %v", err)
+		}
+		// If the cluster does not know about the Console capability, it likely predates 4.12 and we can assume
+		// it has it by default. This is to catch possible future scenarios where we upgrade 4.11 no cap to 4.12 no cap.
+		if !cluster.KnowsCapability(clusterVersion, "Console") ||
+			cluster.HasCapability(clusterVersion, "Console") {
+			if err := createConsoleRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+				return err
+			}
+			if err := createConsoleRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
+				return err
+			}
 		}
 	}
 
 	return nil
 }
 
+func isRouteAvailable(ctx context.Context, config *rest.Config, namespace, name string) (bool, error) {
+	routeClient, err := routeclient.NewForConfig(config)
+	if err != nil {
+		return false, err
+	}
+	_, err = routeClient.RouteV1().Routes(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func isConfigAPIGroupAvailable(config *rest.Config) (bool, error) {
```

Comment:
Worth making the function more generic. E.g. `isAPIGroupAvailable(config *rest.Config, apigroup string)`. Or, more preferable to reuse `exutil.DoesApiResourceExist`.

Reply:
Was not keen on refactoring seemingly unrelated parts of the code, but this change might be small enough to do so.

Reply:
I meant to refactor `isConfigAPIGroupAvailable`. Or call `exutil.DoesApiResourceExist` as it is instead of `isConfigAPIGroupAvailable`. Sorry for the confusion.

Reply:
Done